cogeo_mosaic_tiler/handlers/app.py

```diff
@@ -144,156 +143,95 @@ def _create(
     return ("OK", "application/json", json.dumps(meta))
 
 
-@app.route(
-    "/add",
-    methods=["POST"],
-    cors=True,
-    payload_compression_method="gzip",
-    binary_b64encode=True,
-    tag=["mosaic"],
-)
-def _add(body: str, mosaicid: str = None) -> Tuple[str, str, str]:
-    # TODO: Need validation
-    mosaic_definition = json.loads(body)
-
-    if not mosaicid:
-        mosaicid = get_hash(body=body)
+@app.post("/add", tag=["mosaic"], **params)
+def _add(body: str, mosaicid: str) -> Tuple:
+    if _aws_head_object(_create_mosaic_path(mosaicid)):
+        return ("NOK", "text/plain", f"Mosaic: {mosaicid} already exist.")
 
-    key = f"mosaics/{mosaicid}.json.gz"
-    bucket = os.environ["MOSAIC_DEF_BUCKET"]
-    _aws_put_data(key, bucket, _compress_gz_json(mosaic_definition), client=s3_client)
+    mosaic_definition = MosaicJSON(**json.loads(body))
+    with MosaicBackend(
+        _create_mosaic_path(mosaicid), mosaic_def=mosaic_definition
+    ) as mosaic:
+        mosaic.write(gzip=True)
 
-    return (
-        "OK",
-        "application/json",
-        json.dumps({"id": mosaicid, "url": f"s3://{bucket}/{key}"}),
-    )
+    return ("OK", "application/json", json.dumps({"id": mosaicid, "status": "READY"}))
```

Comment:
⚠️ this is a change, we are not returning the S3 url but just the mosaic id and a status